libraries/chain/abi_serializer.cpp

```diff
@@ -207,6 +207,19 @@ namespace eosio { namespace chain {
       return ends_with(type, "[]");
    }
 
+   bool abi_serializer::is_szarray(const string_view& type)const {
+      auto pos1 = type.find_last_of('[');
+      auto pos2 = type.find_last_of(']');
+      if(pos1 == string_view::npos || pos2 == string_view::npos) return false;
+      auto pos = pos1 + 1;
+      if(pos == pos2) return false;
+      while(pos < pos2) {
+         if( ! (type[pos] >= '0' && type[pos] <= '9') ) return false;
+         ++pos;
+      }
+      return true;
+   }
```

Comment:
`t][`
pos1=2
pos2=1
pos=3
returns true

Reply:
- `is_szarray()` is not using any other members in the `abi_serializer` class, it should be a free function. 
- This can be combined with `is_array()` to be  implemented more efficiently and correctly as
```
std::string_view get_array_type_name(std::string_view type) {
    if (type.back() != ']') return {};
    auto it = std::find_if_not(type.rbegin() + 1, type.rend(), [](char x) { return x >= '0' &&  x <='9'; });
    if (it == type.rend() || *it != '[') return {};
    return {type.begin(),  static_cast<std::size_t>(type.rend() - it - 1)};
}
```